open-api/rest-catalog-open-api.yaml

```diff
@@ -1016,6 +1040,110 @@ components:
               items:
                 type: integer
 
+    Expression:
+      oneOf:
+        - $ref: '#/components/schemas/AndOrExpression'
+        - $ref: '#/components/schemas/NotExpression'
+        - $ref: '#/components/schemas/SetExpression'
+        - $ref: '#/components/schemas/LiteralExpression'
+
+    ExpressionType:
+      type: string
+      example:
+        - "eq"
+        - "and"
+        - "or"
+        - "not"
+        - "in"
+        - "not-in"
+        - "lt"
+        - "lt-eq"
+        - "gt"
+        - "gt-eq"
```

Comment:
For updates, we use a single enum, rather than restricting values in each type:

```
      properties:
        action:
          type: string
          enum:
            - upgrade-format-version
            - add-schema
            - set-current-schema
            - add-spec
            - set-default-spec
            - add-sort-order
            - set-default-sort-order
            - add-snapshot
            - set-snapshot-ref
            - remove-snapshots
            - remove-snapshot-ref
            - set-location
            - set-properties
            - remove-properties
```

It seems like we should be consistent with the approach for type identifiers.

Reply:
@nastra, should we use an enum here?

Reply:
Nevermind, I guess this is using enums in each type.